db/controllers.js

```diff
@@ -10,9 +10,30 @@ function getTopics (req, res) {
 function getApi (req, res) {
     res.status(200).send(endpoints)
 }
+
+function getArticles(req, res, next) {
+    fetchArticles()
+    .then((articles) => {
+        res.status(200).send({articles})
+    })
+    .catch(next);
+}
+        
+
+function getArticleById (req, res, next) {
+    const { article_id } = req.params
+    fetchArticleById(article_id)
+    .then((article) => {
+        if (!article) {
+            next()
+        }
```

Comment:
Instead of dealing with this here, check the rows in the model and then do a Promise.reject. It will give you more control over the error you are returning and dealing with it before you send anything back from the model. The rejected promise will be caught in the catch block below